File: AccessPatternCSharp/AccessPatternCSharp/StudentDAO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AccessPatternCSharp
{
    public class StudentDAO : DAO<Student>
    {
        private List<Student> students = new List<Student>();
        public void Create(Student student)
        {
            students.Add(student);
        }

        public void Delete(Student student)
        {
            students.Remove(student);
        }

        public List<Student> GetAll()
        {
            return students;
        }

        public Student GetById(int id)
        {
            for (int i = 0; i < students.Count; i++)
            {
                if (students[i].GetId() == id)
                {
                    return students[i];
                }
            }
            return null;
        }

        public void Update(Student student)
        {
            for (int i = 0; i < students.Count; i++)
            {
                if (students[i].GetId() == student.GetId())
                {
                    students[i].SetName(student.GetName());
                    students[i].SetAge(student.GetAge());
                    break;
                }
            }
        }
    }
}

File: DataAccessPattern/src/demo/StudentDao.java
package demo;

import java.util.ArrayList;
import java.util.List;

public class StudentDao implements Dao<Student> {
	
	private List<Student> students = new ArrayList<Student>();
	
	
	
	@Override
	public List<Student> getAll() {
		
		return students;
	}

	@Override
	public Student getById(int id) {
		for(int i = 0 ; i < students.size() ; i++) {
			if(students.get(i).getId() == id) {
				return students.get(i);
			}
		}
		return null;
	}

	@Override
	public void create(Student student) {
		students.add(student);
	}
	@Override
	public void update(Student student) {
		for(int i = 0 ;i < students.size() ; i++) {
			if(students.get(i).getId() == student.getId()) {
				students.get(i).setName(student.getName());
				students.get(i).setAge(student.getAge());
				break;
			}
		}
	}

	@Override
	public void delete(Student student) {
		students.remove(student);
	}

	
	 
}
